arbnode/dataposter/data_poster.go

```diff
@@ -389,13 +397,19 @@ func (p *DataPoster) feeAndTipCaps(ctx context.Context, nonce uint64, gasLimit u
 	}
 
 	elapsed := time.Since(dataCreatedAt)
-	// MaxFeeCap = (BacklogOfBatches^2 * UrgencyGWei^2 + TargetPriceGWei) * GWei
-	maxFeeCap :=
-		arbmath.FloatToBig(
-			(float64(arbmath.SquareUint(backlogOfBatches))*
-				arbmath.SquareFloat(config.UrgencyGwei) +
-				config.TargetPriceGwei) *
-				params.GWei)
+	parameters := make(map[string]interface{})
+	parameters["BacklogOfBatches"] = float64(backlogOfBatches)
+	parameters["UrgencyGWei"] = config.UrgencyGwei
+	parameters["ElapsedTime"] = float64(elapsed)
+	parameters["ElapsedTimeBase"] = float64(config.ElapsedTimeBase)
+	parameters["ElapsedTimeImportance"] = config.ElapsedTimeImportance
+	parameters["TargetPriceGWei"] = config.TargetPriceGwei
+	parameters["GWei"] = params.GWei
+	result, err := p.maxFeeCapExpression.Evaluate(parameters)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error evaluating maxFeeCapExpression: %w", err)
+	}
+	maxFeeCap := arbmath.FloatToBig(result.(float64))
```

Comment:
We should type check here:

res, ok := result.(float64)
if !ok {
  return fmt.Errorf("maxFeeCapExpression did not evaluate to float, result: %v\n", res)
}
maxFeeCap := arbmath.FloatToBig(res)

Reply:
actually evaluation of any mathematical formula will typecast the result into float64 by default, even if the supplied parameters are of other types such as int. So we should be good to assume its of float64 type for any valid result of a mathematical formula

Reply:
We explicitly only initialize permitted variables inside parameters like BacklogOfBatches, UrgencyGWei etc. and these were all typecasted to float64 so we never run into the case you mentioned above. 

PS: if the user would input a mathematical formula involving at least one operator (plus or minus etc) the result of expression.Evaluate(params) for an expression like one you provided above would be an error and the code wouldn't panic.

Reply:
I've updated the usage section of the flag to include available variables to avoid confusion